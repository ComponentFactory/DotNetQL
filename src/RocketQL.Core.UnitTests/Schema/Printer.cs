namespace RocketQL.Core.UnitTests.SchemaValidation;

public class Printer : UnitTestBase
{
    [Theory]
    [InlineData("printer-schema.graphql")]
    public void FileSchemaDefault(string filename)
    {
        var schemaText = File.ReadAllText(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "TestFiles", filename));
        var schema = new Schema();
        schema.Add(schemaText);
        schema.Validate();
        string print = schema.Print();
        print.MatchSnapshot();
    }

    [Theory]
    [InlineData("printer-schema.graphql")]
    public void FileSchemaIndent2Spaces(string filename)
    {
        var schemaText = File.ReadAllText(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "TestFiles", filename));
        var schema = new Schema();
        schema.Add(schemaText);
        schema.Validate();
        string print = schema.Print(new SchemaPrintOptions() { IndentCount = 2 });
        print.MatchSnapshot();
    }

    [Theory]
    [InlineData("printer-schema.graphql")]
    public void FileSchemaIndent2Tabs(string filename)
    {
        var schemaText = File.ReadAllText(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "TestFiles", filename));
        var schema = new Schema();
        schema.Add(schemaText);
        schema.Validate();
        string print = schema.Print(new SchemaPrintOptions() { IndentCharacter = PrintIndentCharacter.Tab, IndentCount = 2 });
        print.MatchSnapshot();
    }

    [Theory]
    [InlineData("printer-schema.graphql")]
    public void FileSchemaNoDescriptions(string filename)
    {
        var schemaText = File.ReadAllText(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "TestFiles", filename));
        var schema = new Schema();
        schema.Add(schemaText);
        schema.Validate();
        string print = schema.Print(new SchemaPrintOptions() { PrintDescriptions = false });
        print.MatchSnapshot();
    }

    [Theory]
    [InlineData("printer-schema.graphql")]
    public void FileSchemaShowPredefined(string filename)
    {
        var schemaText = File.ReadAllText(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "TestFiles", filename));
        var schema = new Schema();
        schema.Add(schemaText);
        schema.Validate();
        string print = schema.Print(new SchemaPrintOptions() { PrintPredefined = true });
        print.MatchSnapshot();
    }
}

