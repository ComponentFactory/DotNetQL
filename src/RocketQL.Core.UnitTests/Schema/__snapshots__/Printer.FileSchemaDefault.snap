"schema description"
schema @directiveAll
{
    query: typeAll
}

"""
Query multiline comment that should 
be output over several lines when printee
"""
type typeAll implements interface1 & interface2 @directiveAll
{
    "one description"
    one: Int
    
    """
    two description over
    multiple lines of
    the output text
    """
    two: Int @directiveAll
    
    three
    (
        arg1: Int = 5 @directiveAll
    ): Int
    four
    (
        arg1: compound = { name: "bob", age: 42, alive: true } @directiveAll
    ): Int
}

interface interface1
{
    one: Int
}

interface interface2
{
    two: Int
}

"""
Interface multiline comment that should 
be output over several lines when printee
"""
interface interfaceAll implements interface1 & interface2 @directiveAll
{
    "one description"
    one: Int
    
    """
    two description over
    multiple lines of
    the output text
    """
    two: Int @directiveAll
    
    three
    (
        arg1: Int = 5 @directiveAll
    ): Int
    four
    (
        arg1: compound = { name: "bob", age: 42, alive: true } @directiveAll
    ): Int
}

"Example union"
union unionAll @directiveAll = typeAll

input compound
{
    name: String
    age: Int
    alive: Boolean
}

"""
Input multiline comment that should 
be output over several lines when printee
"""
input inputAll @directiveAll
{
    "one description"
    one: Int
    
    """
    two description over
    multiple lines of
    the output text
    """
    two: Int = 5 @directiveAll
    
    four: compound = { name: "bob", age: 42, alive: true } @directiveAll
}

"Example enumeration"
enum enumAll @directiveAll
{
    "numero 1"
    FIRST @directiveAll
    
    "numero 2"
    SECOND
    
    "numero 3"
    THIRD
}

"Scalar union"
scalar scalarAll @directiveAll

directive @directiveAll repeatable on SCHEMA | SCALAR | OBJECT | FIELD_DEFINITION | ARGUMENT_DEFINITION | INTERFACE | UNION | ENUM | ENUM_VALUE | INPUT_OBJECT | INPUT_FIELD_DEFINITION

